extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileWriter.java

```diff
@@ -45,16 +46,16 @@ public ParquetFileWriter(
             final int pageSize,
             final ByteBufferAllocator allocator,
             final MessageType type,
-            final CompressionCodecName codecName,
+            final String codecName,
             final Map<String, String> extraMetaData) throws IOException {
         this.pageSize = pageSize;
         this.allocator = allocator;
         this.extraMetaData = new HashMap<>(extraMetaData);
         writeChannel = channelsProvider.getWriteChannel(filePath, false); // TODO add support for appending
         this.type = type;
         this.channelsProvider = channelsProvider;
-        CodecFactory codecFactory = new CodecFactory(new Configuration(), pageSize);
-        this.compressor = codecFactory.getCompressor(codecName);
+        DeephavenCodecFactory ccf = new DeephavenCodecFactory(Collections.emptyList());
```

Comment:
ditto, should probably be singleton, so the configured classes are shared.

Reply:
Note however that it has no state, except the CompressionCodecFactory, and that only has state of the codecs it was built with, no other state maintained along the way - perhaps default ctor instead would pick up the list from Configuration?